.github/workflows/cd.yaml

```diff
@@ -0,0 +1,221 @@
+name: CI
+
+on:
+  release:
+    types: [ published ]
+
+jobs:
+
+  info:
+    name: Prepare release
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release'
+    timeout-minutes: 5
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+    steps:
+      - name: Checkout commit
+        uses: actions/checkout@v2
+      - name: Setup Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: Cache PyPI
+        uses: actions/cache@v2
+        with:
+          key: pip-lint-${{ hashFiles('requirements/*.txt') }}
+          path: ~/.cache/pip
+          restore-keys: |
+              pip-lint-
+      - name: Install dependencies
+        run: |
+          python -m pip
+          make setup
+      - name: Save the package version
+        id: version
+        run: |
+          echo "::set-output name=version::$(python setup.py --version)"
+      - name: Show version
+        run: |
+            echo ${{ steps.version.outputs.version }}
+
+  devpi_deploy:
+    name: Release package
+    runs-on: ubuntu-latest
+    needs: [info]
+    if: github.event_name == 'release'
+    steps:
+      - name: Checkout commit
+        uses: actions/checkout@v2
+      - name: Sanity check for tag and version
+        run: |
+          export VERSION=${{ needs.info.outputs.version }}
+          if [ "${{ github.ref }}" != "refs/tags/$VERSION" ]
+          then
+            echo "Git tag '${{ github.ref }}' differs from hard-coded package version '$VERSION'"
+            exit 1
+          else
+            echo "OK, git tag matches hard-coded package version: '$VERSION'"
+          fi
+      - name: Install python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: Install dependencies
+        run: |
+          python -m pip install twine wheel
+          make setup
+      - name: Make dists
+        run: |
+          python setup.py sdist bdist_wheel
+      - name: PyPI upload
+        env:
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+        run: |
+          twine upload dist/*
+
+  dockerhub_deploy:
+    name: Release image if needed
+    runs-on: ubuntu-latest
+    needs: [info, devpi_deploy]
+    if: github.event_name == 'release'
+    env:
+      DOCKER_SERVER: docker.io
+      DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+      DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+    steps:
+      - name: Checkout commit
+        uses: actions/checkout@v2
+      - name: Install python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: Install dependencies
+        run: |
+          python -m pip install twine wheel
```

Comment:
Do we need twine and wheel here? Pip should be enough.

Reply:
you're right, it's not necessary here.